pallets/proofs-dealer/src/utils.rs

```diff
@@ -413,8 +413,14 @@ where
             seed,
         });
 
-        // Calculate if this is a checkpoint challenge round.
         let last_checkpoint_tick = LastCheckpointTick::<T>::get();
+
+        // Count last checkpoint challenges tick challenges
+        let checkpoint_challenges_count =
+            TickToCheckpointChallenges::<T>::get(last_checkpoint_tick)
+                .unwrap_or_default()
+                .len();
```

Comment:
I need to query it before hand and set to 0 since I would either have to clone the bounded vec for every iteration of slashable providers or read from storage for every iteration. The optimal solution is to read from storage first with potentially 0 as the default without any need of clone or re-reads.

Though I added a check to see if the count is 0. So at least that makes it clear and avoids execution if there aren’t any.